### PR TITLE
Notification System

### PR DESCRIPTION
**Pull request by EvonneNgXinYi**
Build a notification reminder for user logged when their post is commented

### CODE REVIEW

**Comment by EvonneNgXinYi** [comment]
Hi i have a question want to ask:
If i want to set the **auto-destroy **read notification****, 
for example: an auto-destroy the **read** notification in 30 days,
how should i set it?

**Comment by EvonneNgXinYi** [comment]
ready

**Comment by saobv-2223** [comment]
> Hi i have a question want to ask:
> If i want to set the **auto-destroy **read notification****,
> for example: an auto-destroy the **read** notification in 30 days,
> how should i set it?

You can research cronjob with whenever 
https://github.com/javan/whenever

File: app/assets/stylesheets/custom.scss [hunk 1]
```diff
@@ -229,3 +234,32 @@ input {
...
+
+.noti_index {
+  text-align: center;
+}
+
+.number_unread {
+  margin-top: -34px;
+  margin-left: 17px;
+  color: $white;
+}
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]:
```suggestion
}

```

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -33,6 +34,15 @@ def destroy
 
   private
 
+  def create_notification(post, comment)
+    return if post.user.id == current_user.id
```
Comment (saobv-2223) [inline app/controllers/comments_controller.rb:38]:
```suggestion
    return if post.user.id == current_user.id
    
```

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -33,6 +34,15 @@ def destroy
...
+    Notification.create(user_id: post.user.id,
+                        notified_by_id: current_user.id,
+                        post_id: post.id,
+                        identifier: comment.id,
+                        notice_type: "comment")
```
Comment (saobv-2223) [inline app/controllers/comments_controller.rb:44]: You should build params and call here

File: app/controllers/notifications_controller.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class NotificationsController < ApplicationController
+  def link_through
+  	@notification = Notification.find_by id: params[:id]
```
Comment (saobv-2223) [inline app/controllers/notifications_controller.rb]:
```suggestion
    @notification = Notification.find_by id: params[:id]
```

File: app/helpers/notifications_helper.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+module NotificationsHelper
+end
```
Comment (saobv-2223) [inline app/helpers/notifications_helper.rb]: remove file that never used

File: app/views/notifications/_notification.html.erb [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<li>
+  <%= link_to "#{notification.notified_by.name} has #{notification.notice_type}ed on your post", link_through_path(notification) %>
+	<%= time_ago_in_words notification.created_at %> <%= t ".ago" %>
```
Comment (saobv-2223) [inline app/views/notifications/_notification.html.erb]:
```suggestion
    <%= time_ago_in_words notification.created_at %> <%= t ".ago" %>
```

File: app/views/notifications/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+<dir class="container" >
+	<div class="row" >
+	  <div class="col-md-6 col-md-offset-3" >
+			<ul class="notification-index-list" >
+		  	<%= render @notifications %>
+			</ul>
+	  </div>
+	</div>
+</dir>
```
[the remark this answers is not in the document]
Comment (EvonneNgXinYi) [reply app/views/notifications/index.html.erb]: sorry i cant see this picture. May u resend it

File: app/controllers/comments_controller.rb (hunk 3 above)
Comment (EvonneNgXinYi) [reply app/controllers/comments_controller.rb:44]: Sorry I am a bit confusing...
Should i use JSON params or params permit?

**Comment by EvonneNgXinYi** [comment]
ready

**Comment by EvonneNgXinYi** [comment]
> > Hi i have a question want to ask:
> > If i want to set the **auto-destroy **read notification****,
> > for example: an auto-destroy the **read** notification in 30 days,
> > how should i set it?
> 
> You can research cronjob with whenever
> https://github.com/javan/whenever

@saobv-2223 
sorry, I had read through the link that you given, but I am still not sure how to do the coding. Can you teach me how to do it?

File: app/controllers/comments_controller.rb (hunk 3 above)
Comment (saobv-2223) [reply app/controllers/comments_controller.rb:44]: Ok no problem, you can use current code

File: app/views/notifications/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<% provide :title, t(".notifi_index_title") %>
+<h1><%= t ".heading" %></h1>
+
+<dir class="container" >
+  <div class="row" >
+	<div class="col-md-6 col-md-offset-3" >
+	  <ul class="notification-index-list" >
+		<%= render @notifications %>
```
Comment (saobv-2223) [inline app/views/notifications/index.html.erb]: indenting only 2 spaces

**Review by saobv-2223 — COMMENTED** [review]
> > > Hi i have a question want to ask:
> > > If i want to set the **auto-destroy **read notification****,
> > > for example: an auto-destroy the **read** notification in 30 days,
> > > how should i set it?
> > 
> > 
> > You can research cronjob with whenever
> > https://github.com/javan/whenever
> 
> @saobv-2223
> sorry, I had read through the link that you given, but I am still not sure how to do the coding. Can you teach me how to do it?

I think you should code the main flow of project firstly, in order to run perfectly without any errors. Because crontab is not easy, it required ubuntu OS and other knowledge. I only can give you link document to self research

**Comment by EvonneNgXinYi** [comment]
> > > > Hi i have a question want to ask:
> > > > If i want to set the **auto-destroy **read notification****,
> > > > for example: an auto-destroy the **read** notification in 30 days,
> > > > how should i set it?
> > > 
> > > 
> > > You can research cronjob with whenever
> > > https://github.com/javan/whenever
> > 
> > 
> > @saobv-2223
> > sorry, I had read through the link that you given, but I am still not sure how to do the coding. Can you teach me how to do it?
> 
> I think you should code the main flow of project firstly, in order to run perfectly without any errors. Because crontab is not easy, it required ubuntu OS and other knowledge. I only can give you link document to self research

Thank you for your suggestion. Then i will focus the main part first.

**Comment by EvonneNgXinYi** [comment]
ready

**Review by saobv-2223 — COMMENTED** [review]
Having conflict

**Comment by EvonneNgXinYi** [comment]
ready

**Pull request merged**